code/game/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,182 @@
+
+/obj/machinery/electrolyzer/update_overlays()
+	. = ..()
+
+	if(panel_open)
+		. += "electrolyzer-open"
+
+/obj/machinery/electrolyzer/process()
+	if(!is_operational() && on)
+		on = FALSE
+	if(!on)
+		return PROCESS_KILL
+
+	if(cell && cell.charge > 0)
+		var/turf/L = loc
+		if(!istype(L))
+			if(mode != ELECTROLYZER_MODE_STANDBY)
+				mode = ELECTROLYZER_MODE_STANDBY
+				update_icon()
+			return
+
+		var/datum/gas_mixture/env = L.return_air() //get air from the turf
+		var/datum/gas_mixture/removed = env.remove(0.1 * env.total_moles())
+		removed.assert_gases(/datum/gas/water_vapor, /datum/gas/oxygen, /datum/gas/hydrogen)
+		var/proportion = min(removed.gases[/datum/gas/water_vapor][MOLES], 5)//Works on 5 mols at a time.
+		removed.gases[/datum/gas/water_vapor][MOLES] -= proportion * 2 * workingPower
+		removed.gases[/datum/gas/oxygen][MOLES] += proportion * workingPower
+		removed.gases[/datum/gas/hydrogen][MOLES] += proportion * 2 * workingPower
+		env.merge(removed) //put back the new gases in the turf
+		air_update_turf()
+
+		var/newMode = ELECTROLYZER_MODE_STANDBY
+		if(on)
+			newMode = ELECTROLYZER_MODE_WORKING //change the mode to working if the machine is on
+
+		if(mode != newMode) //check if the mode is set correctly
+			mode = newMode
+			update_icon()
+
+		if(mode == ELECTROLYZER_MODE_STANDBY)
+			return
+
+		cell.use(workingPower / (efficiency + workingPower))
+	else
+		on = FALSE
+		update_icon()
+		return PROCESS_KILL
+
+/obj/machinery/electrolyzer/RefreshParts()
+	var/laser = 0
+	var/cap = 0
+	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
+		laser += M.rating
+	for(var/obj/item/stock_parts/capacitor/M in component_parts)
+		cap += M.rating
+
+	workingPower = laser //used in the amount of moles processed
+
+	efficiency = (cap + 1) * 0.5 //used in the amount of charge in power cell uses
+
+/obj/machinery/electrolyzer/attackby(obj/item/I, mob/user, params)
+	add_fingerprint(user)
+	if(default_unfasten_wrench(user, I))
+		return
+	else if(istype(I, /obj/item/stock_parts/cell))
+		if(panel_open)
+			if(cell)
+				to_chat(user, "<span class='warning'>There is already a power cell inside!</span>")
+				return
+			else if(!user.transferItemToLoc(I, src))
+				return
+			cell = I
+			I.add_fingerprint(usr)
+
+			user.visible_message("<span class='notice'>\The [user] inserts a power cell into \the [src].</span>", "<span class='notice'>You insert the power cell into \the [src].</span>")
+			SStgui.update_uis(src)
+		else
+			to_chat(user, "<span class='warning'>The hatch must be open to insert a power cell!</span>")
+			return
+	else if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		panel_open = !panel_open
+		user.visible_message("<span class='notice'>\The [user] [panel_open ? "opens" : "closes"] the hatch on \the [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the hatch on \the [src].</span>")
+		update_icon()
+	else if(default_deconstruction_crowbar(I))
+		return
+	else
+		return ..()
```

Comment:
Use returns instead of so many `else`

Reply:
how do i use returns instead of `else`?

Reply:
instead of
```cpp
else if(default_deconstruction_crowbar(I))
	return
else
	return ..()
```
You can simply do
```cpp
else if(default_deconstruction_crowbar(I))
	return
return ..()
```
There is no need for an `else` if the option right over it returns.

Also instead of this:
```cpp
else if(I.tool_behaviour == TOOL_SCREWDRIVER)
	panel_open = !panel_open
	user.visible_message("<span class='notice'>\The [user] [panel_open ? "opens" : "closes"] the hatch on \the [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the hatch on \the [src].</span>")
	update_icon()
else if(default_deconstruction_crowbar(I))
```
You can do
```cpp
else if(I.tool_behaviour == TOOL_SCREWDRIVER)
	panel_open = !panel_open
	user.visible_message("<span class='notice'>\The [user] [panel_open ? "opens" : "closes"] the hatch on \the [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the hatch on \the [src].</span>")
	update_icon()
	return
if(default_deconstruction_crowbar(I))
```
As there is no code to be executed after the `else`